kinder/pkg/build/alter/alter.go

```diff
@@ -185,13 +188,80 @@ func (c *Context) prepareBits(bitsInstallers []bits.Installer, bc *bits.BuildCon
 	log.Info("Preparing bits ...")
 
 	for _, b := range bitsInstallers {
-		if err := b.Prepare(bc); err != nil {
+		// prepare bits
+		bits, err := b.Prepare(bc)
+		if err != nil {
 			return errors.Wrap(err, "failed to copy alter bits")
 		}
+
+		// fix the bits in order to match kubeadm/kinder expectations
+		// NB. this is done here so all the bits gets fixes, no matter of the source
+		for k, v := range bits {
+			// if the bit is one of the kubernetes images, we should ensure the repository/name matches kubeadm expectations
+			if contains(extract.AllKubernetesImages, k) {
+				if err := fixImageTar(v); err != nil {
+					return errors.Wrap(err, "failed to fix bits")
+				}
+			}
+		}
 	}
+
 	return nil
 }
 
+// fixImageTar ensure the repository/name matches kubeadm expectations
+func fixImageTar(v string) error {
+	log.Infof("fixing %s", v)
+
+	// prepare to read the image tar
+	f, err := os.Open(v)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	// read the image tar and write the fixed version on a string builder
+	var w strings.Builder
+	err = kinddocker.EditArchiveRepositories(f, &w, fixRepository)
+	if err != nil {
+		return err
+	}
+	f.Close()
+
+	// override the image tar with the fixed version
+	err = ioutil.WriteFile(v, []byte(w.String()), 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// fixRepository drop the arch suffix from images to get the expected image;
+// this is necessary for kubernetes v1.15+
+// Nb. for < v1.12 it was requested to do the opposite, but it not necessary anymore
+// because v.11 is already out of the kubeadm e2e test matrix
+func fixRepository(repository string) string {
+	archSuffix := "-amd64"
+
+	if strings.HasSuffix(repository, archSuffix) {
+		fixed := strings.TrimSuffix(repository, archSuffix)
+		fmt.Println("fixed: " + repository + " -> " + fixed)
+		repository = fixed
+	}
+
+	return repository
+}
+
+func contains(s []string, e string) bool {
```

Comment:
possibly better to move this is a closure under `prepareBits` if it's used only there, or leave it package wider but rename it to something more descriptive - e.g. `stringSliceContainsString`